natapokie/ai-fashion-mirror
# 24 add GitHub actions

## Description

**Fixes #24 **

Adding Github Actions -- will build server and client when new changes are pushed to check if things are compiling.
- client builds production app
- server runs jest test (currently a simple test to make sure backend is running)

## Documentation Used

Please list any documentation or resources that were referenced while making these changes.

- [ ] [Github Actions Starter](https://www.youtube.com/watch?v=ylEy4eLdhFs)
- [ ] [checkout task](https://github.com/actions/checkout)
- [ ] [using Node](https://github.com/actions/setup-node)
- [ ] [caching node modules example](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action)
- [ ] [trigger builds on specific paths](https://stackoverflow.com/questions/63822219/how-to-run-github-actions-workflow-only-if-the-pushed-files-are-in-a-specific-fo)
- [ ] [server side testing](https://www.freecodecamp.org/news/how-to-test-in-express-and-mongoose-apps/)
- [ ] [another server side testing doc](https://dev.to/lukekyl/testing-your-express-js-backend-server-3ae6)
- [ ] [jest with typescript and babel](https://jestjs.io/docs/getting-started#using-typescript)

New packages added include [jest](https://www.npmjs.com/package/jest?activeTab=readme), [supertest](https://www.npmjs.com/package/supertest) and babel.
- jest: framework used to create unit tests
- supertest: used to test endpoints and routes on HTTP servers.
- babel: (not really sure) but babel was chosen for Which provider should be used to instrument code for coverage? when setting up jest
- other packages were installed so jest can work with typescript and mainly eslint

## How To Test?

Make sure to run `npm install` on the server side to run the unit tests, unit test can be run using `npm run test`, this will run the test.js files in the tests folder.
Otherwise, for all future prs, you should notice a client/server check depending on what files you modified.

## Screenshots (if applicable):

Checks must succeed before pr can be merged,
![image](https://github.com/user-attachments/assets/ce57fb14-c125-4f41-8dcc-06e1824ceb40)
